src/ir/ty.rs

```diff
@@ -101,6 +129,25 @@ pub trait TemplateDeclaration {
                  .collect())
         }
     }
+
+    /// Get only the set of template parameters that this item uses. This is a
+    /// subset of `all_template_params` and does not necessarily contain any of
+    /// `self_template_params`.
+    fn used_template_params(&self, ctx: &BindgenContext) -> Option<Vec<ItemId>>
```

Comment:
Is the option needed? I believe in most cases it'd be more convenient to just return a `Vec<ItemId>`, and make an empty vector represent the lack of them?